### PR TITLE
fix: ensure clients and server are always on the same version

### PR DESCRIPTION
**Pull request by jamil314**
Solves #6695

### CODE REVIEW

**Comment by jamil314** [comment]
When new content in client is available:
When new content in client is available



When client version does not match gateway version and no new content is available in client:
modal
error

**Comment by jamil314** [comment]
[country config pr](https://github.com/opencrvs/opencrvs-countryconfig/pull/236)

File: packages/client/src/utils/apolloClient.ts [hunk 1]
```diff
@@ -71,12 +72,35 @@ export const createClient = (
...
+  const responseLink = new ApolloLink((operation, forward) => {
+    return forward(operation).map((response) => {
+      const context = operation.getContext()
+      const {
+        response: { headers }
+      } = context
+
+      const gatewayVersion = headers.get('X-version')
+      if (gatewayVersion !== APP_VERSION) {
+        // const errorMsg = `Version Mismatch: Register Client is running on ${APP_VERSION}, whereas gateway is running on ${gatewayVersion}. Please reload to get the latest client`
```
Comment (rikukissa) [inline packages/client/src/utils/apolloClient.ts:84]:
```suggestion
```

File: packages/login/src/utils/authApi.ts [hunk 2]
```diff
@@ -74,12 +74,29 @@ export interface ITokenResponse {
...
+
 export function request<T>(options: AxiosRequestConfig) {
   const onSuccess = (response: AxiosResponse<T>) => {
+    const gatewayVersion = response.headers['x-version']
+
+    if (gatewayVersion && gatewayVersion !== loginClientVersion) {
+      console.log(
+        `Version Mismatch: Frontend is running on ${loginClientVersion}, whereas backend is running on ${gatewayVersion}. Please Reload to get the latest client`
+      )
+      throw new AxiosError('Version_MisMatch')
```
Comment (rikukissa) [inline packages/login/src/utils/authApi.ts:87]: Can we use `VERSION_MISMATCH` instead? It's a bit more standard capitalisation for a constant error identifier like this

**Comment by jamil314** [comment]
[country config translations](https://github.com/opencrvs/opencrvs-countryconfig/pull/242)

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@
 - Supporting document fields can now be made required
 - If there is only one option in the document uploader select, then it stays hidden and only the upload button is showed with the only option being selected by default
 - Allow configuring the default search criteria for record search [#6924](https://github.com/opencrvs/opencrvs-core/issues/6924)
+- Ensure client and server are always on the same version [#6695](https://github.com/opencrvs/opencrvs-core/issues/6695)
```
Comment (rikukissa) [inline CHANGELOG.md:16]:
```suggestion
- Add checks to validate client and server are always on the same version. This prevents browsers with a cached or outdated client versions from making potentially invalid requests to the backend [#6695](https://github.com/opencrvs/opencrvs-core/issues/6695)
```

**Review by rikukissa — APPROVED** [review]

**Pull request merged**